File: app/src/main/java/com/vpn/supervpnfree/MainApp.java
package com.vpn.supervpnfree;

import android.annotation.SuppressLint;
import android.content.Context;
import androidx.multidex.MultiDexApplication;

import com.tencent.mmkv.MMKV;
import com.vpn.supervpnfree.data.Hot;
import com.vpn.supervpnfree.updata.UpDataUtils;
import com.vpn.supervpnfree.utils.AdManager;
import com.vpn.supervpnfree.utils.GlobalTimer;

public class MainApp extends MultiDexApplication {

    @SuppressLint("StaticFieldLeak")
    private static Context context;
    @SuppressLint("StaticFieldLeak")
    private static MainApp mAppInstance;

    public static Context getContext() {
        return context;
    }

    public static synchronized MainApp getAppInstance() {
        return mAppInstance;
    }

    public static AdManager adManager;
    public static GlobalTimer globalTimer;

    public static MMKV saveLoadManager;

    @Override
    public void onCreate() {
        super.onCreate();
        mAppInstance = this;
        context = this;
        initHydraSdk();
        Hot.INSTANCE.initCore(this);
        if(Hot.INSTANCE.isMainProcess(this)){
            adManager = new AdManager(this);
            Hot.INSTANCE.adAndFirebaseBase(this);
            Hot.INSTANCE.registerAppLifeCallback(this);
            globalTimer = new GlobalTimer();
        }
    }


    public void initHydraSdk() {
        MMKV.initialize(this);
        saveLoadManager =
                MMKV.mmkvWithID("EasyVPN", MMKV.MULTI_PROCESS_MODE);
    }


}

File: app/src/main/java/com/vpn/supervpnfree/MainApp.kt
package com.vpn.supervpnfree

import android.annotation.SuppressLint
import android.app.Application
import android.content.Context
import com.github.shadowsocks.Core
import com.google.android.gms.ads.MobileAds
import com.google.firebase.FirebaseApp
import com.google.firebase.ktx.Firebase
import com.google.firebase.ktx.initialize
import com.tencent.mmkv.MMKV
import com.vpn.supervpnfree.activities.MainActivity
import com.vpn.supervpnfree.data.Hot.isMainProcess
import com.vpn.supervpnfree.data.Hot.registerAppLifeCallback
import com.vpn.supervpnfree.utils.AdManager
import com.vpn.supervpnfree.utils.GlobalTimer

class MainApp : Application() {
    override fun onCreate() {
        super.onCreate()
        Firebase.initialize(this)
        FirebaseApp.initializeApp(this)
        context = this
        initHydraSdk()
        Core.init(this, MainActivity::class)
        if (isMainProcess(this)) {
            MobileAds.initialize(this)
            adManager = AdManager(this)
            registerAppLifeCallback(this)
            globalTimer = GlobalTimer()
        }
    }
    fun initHydraSdk() {
        MMKV.initialize(this)
        saveLoadManager =
            MMKV.mmkvWithID("EasyVPN", MMKV.MULTI_PROCESS_MODE)
    }


    companion object {
        lateinit var context: Context
        lateinit var adManager: AdManager
        lateinit var globalTimer: GlobalTimer
        lateinit var saveLoadManager: MMKV
    }
}
